src/use-localstorage.ts

```diff
@@ -72,20 +79,13 @@ export function useLocalStorage<TValue = string>(key: string, initialValue?: TVa
     }
 
     return () => {
-      window.removeEventListener(
-        LocalStorageChanged.eventName,
-        (e: any) => onLocalStorageChange(e as LocalStorageChanged<TValue>)
-      );
-      window.removeEventListener('storage', e => onLocalStorageChange(e));
+      window.removeEventListener(LocalStorageChanged.eventName, listener);
```

Comment:
Now, when the listener is passed to `removeEventListener` it is referentially equal and actually removes it.